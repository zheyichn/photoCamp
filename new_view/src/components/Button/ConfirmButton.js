import { Button } from "@mui/material";
import { confirmButtonStyle } from "../../styles/buttonStyles";

export default function ConfirmButton(props) {
  return (
    <Button
      data-testid="confirm-btn"
      variant="text"
      disabled={props.disabled}
      onClick={props.onClick}
      sx={confirmButtonStyle}
    >
      Confirm
    </Button>
  );
}
