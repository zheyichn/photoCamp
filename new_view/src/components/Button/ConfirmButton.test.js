import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import ConfirmButton from "./ConfirmButton";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";

afterEach(() => {
  cleanup();
});

describe("ConfirmButton component renders correctly", () => {
  test("confirm button displayed correctly", () => {
    render(<ConfirmButton />);
    const confirmButton = screen.getByTestId("confirm-btn");
    expect(confirmButton).toBeInTheDocument();
  });

  test("confirm button works appropriately", () => {
    const mockCancelFunction = jest.fn();
    render(<ConfirmButton onClick={mockCancelFunction} />);
    const confirmButton = screen.getByTestId("confirm-btn");
    userEvent.click(confirmButton);
    expect(mockCancelFunction).toHaveBeenCalled();
  });

  it("cancel button match snapshots", () => {
    const tree = renderer.create(<ConfirmButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
