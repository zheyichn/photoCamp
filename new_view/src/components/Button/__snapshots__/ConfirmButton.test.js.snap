// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmButton component renders correctly cancel button match snapshots 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1fxqi4r-MuiButtonBase-root-MuiButton-root"
  data-testid="confirm-btn"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Confirm
</button>
`;
